allow generate self play with other model specified in arg

amy-cao opened the pull request:
add argparser to allow 'python train.py --argument'
allow generate self play between current model and another model given the path specified in the argument

amy-cao committed:
add argparser to allow 'python train.py --argument'
allow generate self play between current model and another model given the path specified in the argument

kenziyuliu commented on train.py:
```diff
@@ -367,20 +371,36 @@ def get_rand_prev_version(upper):
 
+def build_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--model_path',
```
add short hand flags e.g. "-m" for "--modelpath"

kenziyuliu replied:
same for other arguments

kenziyuliu commented on train.py:
```diff
@@ -367,20 +371,36 @@ def get_rand_prev_version(upper):
+        ITERATION_COUNT = 0
+
+    BEST_MODEL = args.best_model_path
+    if BEST_MODEL is not None:
```
Unify this (global variables) and the argument parser; use one of them only

kenziyuliu commented on train.py:
```diff
@@ -235,7 +236,7 @@ def evaluate_in_parallel(best_model, cur_model, num_games, num_workers):
 
 
-def evolve(cur_model_path):
+def evolve(cur_model_path, other_opponent_for_selfplay):
```
since we can take best model/other opponent/etc, maybe design a dictionary for passing in these, and document the keys somewhere

amy-cao committed:
-remove global variable ITERATION_COUNT and BEST_MODEL
-add shorter argument parsers -p,-c,-b for opponent, current, best model respectively
